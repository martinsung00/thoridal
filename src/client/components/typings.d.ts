// Assists in building webpack with CSS modules & SVG (as ReactComponent)

declare module "*.css" {
  interface IClassNames {
    [className: string]: string;
  }
  const classNames: IClassNames;
  export = classNames;
}

declare module "*.svg" {
  import React = require("react");
  export const ReactComponent: React.SFC<React.SVGProps<SVGSVGElement>>;
  const src: string;
  export = src;
}
