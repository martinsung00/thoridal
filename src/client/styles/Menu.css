.root {
  height: 32em;
  width: 95%;
  position: relative;
  left: 2.25%;
  margin-top: 8%;
  margin-bottom: 5%;
}
