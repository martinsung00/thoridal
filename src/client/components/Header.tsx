import React from "react";
import styles from "../styles/Header.css";
import logo from "../logo/logo.svg";
import spyglass from "../logo/spyglass.svg";

interface Props {
  // To-do: Specify prop types
}

interface State {
  // To-do: Add more state types
  searchType: string;
  searchTerm: string | undefined;
}

export default class Header extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      searchType: "id",
      searchTerm: "",
    };
  }

  _handleChange(event: React.ChangeEvent<HTMLInputElement>): void {
    this.setState({
      searchTerm: event.target.value,
    });
  }

  _handleSelect(event: React.FormEvent<HTMLSelectElement>) {
    this.setState({
      searchType: event.currentTarget.value,
    });
  }

  render() {
    return (
      <div>
        <div className={styles.head}>
          <div className={styles.title}>THORIDAL</div>
          <img src={logo} alt="Main Logo" className={styles.mainLogo} />
          <div className={styles.searchbar}>
            <img
              src={spyglass}
              alt="Search Icon"
              className={styles.searchIcon}
            />
            <div>
              <input
                type="text"
                placeholder="Search By"
                value={this.state.searchTerm}
                onChange={this._handleChange.bind(this)}
                className={styles.search}
              />
            </div>
            <div className={styles.searchType}>
              <select
                className={styles.sortBy}
                value={this.state.searchType}
                onChange={this._handleSelect.bind(this)}
                defaultValue="id"
              >
                <option value="id">ID</option>
                <option value="company">Company</option>
                <option value="ticker">Ticker</option>
                <option value="date">Date</option>
              </select>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
