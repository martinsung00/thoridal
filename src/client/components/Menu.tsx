import React from "react";
import styles from "../styles/Menu.css";
import { Add } from "./menu_components/MenuAdd";
import { Delete } from "./menu_components/MenuDelete";
import { Update } from "./menu_components/MenuUpdate";
import { Search } from "./menu_components/MenuSearch";

interface Props {
  // To-do add props as needed
}

export const Menu: React.FunctionComponent<Props> = () => {
  return (
    <div className={styles.root}>
      <Add />
      <Search />
      <Delete />
      <Update />
    </div>
  );
};
