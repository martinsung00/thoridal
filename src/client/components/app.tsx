import * as React from "react";
import { Trade } from "../../server/types/trade";
import Header from "./Header";

interface Props {
  // To-do: Specify prop types
}

interface State {
  // To-do: Add more state types
  previousWeek: Array<Trade>;
}

export default class App extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      previousWeek: [],
    };
  }

  render() {
    return (
      <div>
        <Header />
      </div>
    );
  }
}
