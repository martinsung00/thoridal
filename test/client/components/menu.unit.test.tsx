import React from "react";
import renderer from "react-test-renderer";
import { Menu } from "../../../src/client/components/Menu";

describe("Menu Tests", function () {
  it("should render as is", function () {
    const container = renderer.create(<Menu />).toJSON();
    expect(container).toMatchSnapshot();
  });
});
