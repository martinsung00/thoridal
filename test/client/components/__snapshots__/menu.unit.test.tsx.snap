// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Tests should render as is 1`] = `
<div>
  <div>
    <div
      id="menu-add"
      onClick={[Function]}
    >
      <div>
        Write a Trade
      </div>
      <div />
      <div />
      <div />
    </div>
  </div>
  <div>
    <div
      id="menu-search"
      onClick={[Function]}
    >
      <div>
        Find a Trade
      </div>
      <div />
      <div />
    </div>
  </div>
  <div>
    <div
      id="menu-delete"
      onClick={[Function]}
    >
      <div>
        Delete a Trade
      </div>
      <div />
      <div>
        Ã—
      </div>
    </div>
  </div>
  <div>
    <div
      id="menu-update"
      onClick={[Function]}
    >
      <div>
        Update a Trade
      </div>
      <div>
        <div />
        <div />
      </div>
      <div>
        <div />
        <div />
      </div>
      <div>
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
